### PR TITLE
GUI Window Resources

### PR DESCRIPTION
**Pull request by gralkapk**
Resource interface for GUI windows.

### CODE REVIEW

File: frontend/services/gui/src/GUI_Service.hpp [hunk 1]
```diff
@@ -84,8 +84,8 @@ class GUI_Service final : public AbstractFrontendService {
     std::vector<std::string> m_queuedProjectFiles;
 
     std::vector<FrontendResource> m_providedResourceReferences;
-    std::vector<FrontendResource> m_requestedResourceReferences;
     std::vector<std::string> m_requestedResourcesNames;
+    std::shared_ptr<frontend_resources::FrontendResourcesMap> frontend_resources;
```
Comment (geringsj) [inline frontend/services/gui/src/GUI_Service.hpp:89]: There is no strict need to use a `std::shared_ptr` here. Nothing bad happens if `FrontendResourcesMap` gets copied, everything has value semantics.

Comment (gralkapk) [reply frontend/services/gui/src/GUI_Service.hpp:89]: The `FrontendResourceMap` carries data that would be copied to every single window though.

**Comment by geringsj** [comment]
The `std::map` contained in the `FrontendResourcesMap` should be fine for copies and the contained resource wrappers are also pretty dumb, the idea beeing that different participants can keep a hard copy of their resource (references) map for convenience, since resources will not change during runtime. This may help maintain isolation of resources such that participants dont (accidentally) get access to resources they did not request. Just trying to give a little context.

**Review by reinago — APPROVED** [review]

**Pull request merged**